File: src/multimodal/universalMultimodalExtractionEngine.ts
/**
 * Converted from Python: core/multimodal/universal_multimodal_extraction_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { extractOcr } from "../ocr/ocrEngine.js";
import { detectLayoutBlocks } from "../layout/layoutDetectionEngine.js";
import { extractTables } from "../tables/tableExtractionEngine.js";
import { extractForms } from "../vision/formExtractionEngine.js";
import { detectCharts } from "../vision/chartDetectionEngine.js";
import { detectUiComponents } from "../vision/uiComponentDetectionEngine.js";
import { compileMultimodalIr } from "../ir/multimodalIr.js";

export function extractMultimodal(path: any): any {
  var ocr: any = extractOcr(path);
  var layout: any = detectLayoutBlocks(py.get(ocr, "regions", []));
  var tables: any = extractTables(layout);
  var forms: any = extractForms(layout);
  var charts: any = detectCharts(layout);
  var ui: any = detectUiComponents(layout);
  var multimodal_ir: any = compileMultimodalIr(layout, tables, forms, charts, ui);
  return {"ocr": ocr, "layout": layout, "tables": tables, "forms": forms, "charts": charts, "ui": ui, "multimodal_ir": multimodal_ir, "bounded": true};
}
export { compileMultimodalIr, detectCharts, detectLayoutBlocks, detectUiComponents, extractForms, extractOcr, extractTables };

File: lib/src/multimodal/universal_multimodal_extraction_engine.dart
/// Port of core/multimodal/universal_multimodal_extraction_engine.py:
/// extract_multimodal.
library;

import 'package:webweavex/src/ir/multimodal_ir.dart';
import 'package:webweavex/src/layout/layout_detection_engine.dart';
import 'package:webweavex/src/ocr/ocr_engine.dart';
import 'package:webweavex/src/tables/table_extraction_engine.dart';
import 'package:webweavex/src/vision/chart_detection_engine.dart';
import 'package:webweavex/src/vision/form_extraction_engine.dart';
import 'package:webweavex/src/vision/ui_component_detection_engine.dart';

/// Port of universal_multimodal_extraction_engine.extract_multimodal.
Map<String, dynamic> extractMultimodal(
  String path,
) {
  final Map<String, dynamic> ocr = extractOcr(path);

  final List<dynamic> regionsRaw = ocr.containsKey('regions')
      ? ocr['regions'] as List<dynamic>
      : <dynamic>[];
  final List<Map<String, dynamic>> regions = <Map<String, dynamic>>[
    for (final dynamic region in regionsRaw) region as Map<String, dynamic>,
  ];

  final Map<String, dynamic> layout = detectLayoutBlocks(regions);

  final Map<String, dynamic> tables = extractTables(layout);

  final Map<String, dynamic> forms = extractForms(layout);

  final Map<String, dynamic> charts = detectCharts(layout);

  final Map<String, dynamic> ui = detectUiComponents(layout);

  final Map<String, dynamic> multimodalIr = compileMultimodalIr(
    layout: layout,
    tables: tables,
    forms: forms,
    charts: charts,
    ui: ui,
  );

  return <String, dynamic>{
    'ocr': ocr,
    'layout': layout,
    'tables': tables,
    'forms': forms,
    'charts': charts,
    'ui': ui,
    'multimodal_ir': multimodalIr,
    'bounded': true,
  };
}
